sdk/eidolon_ai_sdk/cpu/conversational_apu.py

```diff
@@ -180,11 +182,23 @@ async def _llm_execution_cycle(
             if stream_collector.get_content():
                 logger.info(f"LLM Response: {stream_collector.get_content()}")
 
-            assistant_message = AssistantMessage(
-                type="assistant",
-                content=stream_collector.get_content() or "",
-                tool_calls=[tce.tool_call for tce in tool_call_events],
-            )
+            if isinstance(self.llm_unit, ToolCallLLMWrapper) and tool_call_events:
```

Comment:
The apu shouldn't be aware of this.

Messages need to be reported back as tool calls, and translated by the ToolCallLLMWrapper.

Reply:
Wasn't able to make the meeting today either, but looking here now. So the APU shouldn't try to see if what it got a message from was one type of thing or another thing? ... That it needs to be a more generic interface?

Reply:
@atljoseph , the apu can be aware of message types, but it shouldn't be aware of the impl of the llm unit. The fact that the llm_unit impl requires the messages to be transformed back and forth is a detail of the llm unit impl that the apu should not be aware of